templates/ikfast61_moveit_plugin_template.cpp

```diff
@@ -370,6 +423,40 @@ bool IKFastKinematicsPlugin::initialize(const std::string &robot_description,
   return true;
 }
 
+void IKFastKinematicsPlugin::setSearchDiscretization(const std::map<int,double>& discretization)
+{
+
+  if(discretization.empty())
+  {
+    ROS_ERROR("The 'discretization' map is empty");
+    return;
+  }
+
+  if(redundant_joint_indices_.empty())
+  {
+    ROS_ERROR_STREAM("This group's solver doesn't support redundant joints");
+    return;
+  }
+
+  if(discretization.begin()->first != redundant_joint_indices_[0])
+  {
+    std::string redundant_joint = joint_names_[free_params_[0]];
+    ROS_ERROR_STREAM("Attempted to discretize a non-redundant joint "<<discretization.begin()->first<<", only joint '"<<
+                     redundant_joint<<"' with index " <<redundant_joint_indices_[0]<<" is redundant.");
+    return;
+  }
+
+  redundant_joint_discretization_.clear();
```

Comment:
Another error check should be added here to ensure the discretization is never zero, otherwise a divide by zero error will happen at the variable ``joint_dscrt`` further down in the code